pkg/core/resources/apis/meshexternalservice/api/v1alpha1/helpers.go

```diff
@@ -59,3 +64,39 @@ func (t *MeshExternalServiceResource) Domains() *xds_types.VIPDomains {
 	}
 	return nil
 }
+
+func (t *MeshExternalServiceResource) GetPorts() []core.Port {
+	return []core.Port{
+		MesPort{
+			Value:    t.Spec.Match.Port,
+			Protocol: t.Spec.Match.Protocol,
+		},
+	}
+}
+
+func (t *MeshExternalServiceResource) FindPortByName(name string) (core.Port, bool) {
+	if name == strconv.Itoa(int(t.Spec.Match.Port)) {
+		return MesPort{
+			Value:    t.Spec.Match.Port,
+			Protocol: t.Spec.Match.Protocol,
+		}, true
+	}
+	return MesPort{}, false
+}
+
+type MesPort struct {
```

Comment:
Why redefine a type instead of adding the methods on the existing `t.Spec.Match` ?

Reply:
I was thinking of just implementing interface on existing `t.Spec.Match` but @lobkovilya said that we should add dedicated type for it

Reply:
Any reasons other than just using more memory? We do this for other data types.